src/main/resources/com/google/api/codegen/ruby/main.snip

```diff
@@ -378,44 +378,57 @@
        sample = comments(util.getDocLines(sampleCode(methodView)))
     @if documentation
       {@documentation}
+      @if method.getRequestStreaming()
+        @#
+        @# EXPERIMENTAL: This method interface might change in the future.
+        @#
+      @end
     @end
 
     @if sample
       {@sample}
     @end
 
-    @if or(requiredParams, optionalParams)
+    @if method.getRequestStreaming()
       def {@methodName} @\
-          @if requiredParams
-            @if optionalParams
-              {@requiredParameterNames(requiredParams)},
-              {@optionalParameterValues(optionalParams)},
+          reqs,
+          options: nil
+          @@{@methodName}.call(reqs, options)
+      end
+    @else
+      @if or(requiredParams, optionalParams)
+        def {@methodName} @\
+            @if requiredParams
+              @if optionalParams
+                {@requiredParameterNames(requiredParams)},
+                {@optionalParameterValues(optionalParams)},
+              @else
+                {@requiredParameterNames(requiredParams)},
+              @end
             @else
-              {@requiredParameterNames(requiredParams)},
+              {@optionalParameterValues(optionalParams)},
             @end
```

Comment:
I don't think you edited this but it might be good to swap this logic to be a little bit cleaner.

ie 
Swap
```snip
      def {@methodName} @\
          @if requiredParams
            @if optionalParams
              {@requiredParameterNames(requiredParams)},
              {@optionalParameterValues(optionalParams)},
            @else
              {@requiredParameterNames(requiredParams)},
            @end
          @else
            {@optionalParameterValues(optionalParams)},
          @end
          options: nil
```
for this.
```snip
      def {@methodName} @\
          @if requiredParams
            {@requiredParameterNames(requiredParams)},
          @end
          @if optionalParams
            {@optionalParameterValues(optionalParams)},
          @end
          options: nil
```